### PR TITLE
FEATURE: Make rendering of request in exception log configurable

### PR DESCRIPTION
**Pull request by albe**
With this it is possible to configure if the request data should be rendered into exception stack traces with the newly added configuration setting `Neos.Flow.log.throwables.renderRequestInformation`.

Related #2360

### CODE REVIEW

File: Neos.Flow/Resources/Private/Schema/Settings/Neos.Flow.log.schema.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ properties:
   throwables:
     type: dictionary
     properties:
+      renderRequestInformation: { type: boolean }
       fileStorage:
```
Comment (albe) [inline Neos.Flow/Resources/Private/Schema/Settings/Neos.Flow.log.schema.yaml:7]: I noticed that this schema seems way off from the actual settings above?

File: Neos.Flow/Classes/Core/Booting/Scripts.php [hunk 2]
```diff
@@ -268,17 +268,16 @@ public static function initializeSystemLogger(Bootstrap $bootstrap)
...
      */
-    protected static function initializeExceptionStorage(Bootstrap $bootstrap): ThrowableStorageInterface
+    protected static function initializeExceptionStorage(Bootstrap $bootstrap, array $settings): ThrowableStorageInterface
     {
-        $configurationManager = $bootstrap->getEarlyInstance(ConfigurationManager::class);
-        $settings = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_SETTINGS, 'Neos.Flow');
-
         $storageClassName = $settings['log']['throwables']['storageClass'] ?? FileStorage::class;
         $storageOptions = $settings['log']['throwables']['optionsByImplementation'][$storageClassName] ?? [];
+        $renderRequest = $settings['log']['throwables']['renderRequestInformation'] ?? true;
```
Comment (sorenmalling) [inline Neos.Flow/Classes/Core/Booting/Scripts.php:280]: If this way, lets name the variable "renderRequestInformation" to honour the full Settings key

**Comment by sorenmalling** [comment]
@albe Very good idea - should we do solution discussion in a issue ticket as reference for coming release? 

I would like to look into a way to filter out parameters/header/body-data from log data, so it's not "yes or no" but more "yes, but only these fields"

**Comment by albe** [comment]
Yes, that would be preferable. This is basically the low-hanging-fruit (read: simple) solution to the problem :) +1 for creating an issue that describes the background

**Comment by sorenmalling** [comment]
> Yes, that would be preferable. This is basically the low-hanging-fruit (read: simple) solution to the problem :) +1 for discussing fine details in the issue

+1 for low hanging fruit - let's keep the issue open afterwards, if this change is merged in 👍

**Comment by albe** [comment]
> > Yes, that would be preferable. This is basically the low-hanging-fruit (read: simple) solution to the problem :) +1 for discussing fine details in the issue
> 
> +1 for low hanging fruit - let's keep the issue open afterwards, if this change is merged in 👍

That's why I chose "Related" instead of "Resolves" in the PR message :)

File: Neos.Flow/Classes/Core/Booting/Scripts.php (hunk 2 above)
Comment (albe) [inline Neos.Flow/Classes/Core/Booting/Scripts.php:280]:
```suggestion
        $renderRequestInformation = $settings['log']['throwables']['renderRequestInformation'] ?? true;
```

File: Neos.Flow/Classes/Core/Booting/Scripts.php [hunk 3]
```diff
@@ -295,7 +294,7 @@ protected static function initializeExceptionStorage(Bootstrap $bootstrap): Thro
             return Debugger::getBacktraceCode($backtrace, false, true);
         });
 
-        $throwableStorage->setRequestInformationRenderer(function () {
+        $throwableStorage->setRequestInformationRenderer(function () use ($renderRequest) {
```
Comment (albe) [inline Neos.Flow/Classes/Core/Booting/Scripts.php:297]:
```suggestion
        $throwableStorage->setRequestInformationRenderer(function () use ($renderRequestInformation) {
```

File: Neos.Flow/Classes/Core/Booting/Scripts.php [hunk 4]
```diff
@@ -310,7 +309,9 @@ protected static function initializeExceptionStorage(Bootstrap $bootstrap): Thro
 
             $request = $requestHandler->getHttpRequest();
             $response = $requestHandler->getHttpResponse();
-            $output .= PHP_EOL . 'HTTP REQUEST:' . PHP_EOL . ($request == '' ? '[request was empty]' : $request) . PHP_EOL;
+            if ($renderRequest) {
```
Comment (albe) [inline Neos.Flow/Classes/Core/Booting/Scripts.php:312]:
```suggestion
            if ($renderRequestInformation) {
```

**Review by bwaidelich — COMMENTED** [review]
Makes sense and looks good (by reading) but IMO the new configuration option should be documented at https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartIII/Logging.html

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Logging.rst [hunk 5]
```diff
@@ -107,16 +107,18 @@ You might already have stumbled across such an exception (though hopefully not!)
   HTTP REQUEST:
   127.0.0.1:8081keep-aliveno-cacheno-cacheimageMozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.361image/webp,image/apng,image/*,*/*;q=0.8same-originno-corshttp://127.0.0.1:8081/flow/welcomegzip, deflate, brde-DE,de;q=0.9,en-US;q=0.8,en;q=0.7
 
-  HTTP RESPONSE:
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Logging.rst:110]: The response is no longer logged as of Flow 7 (and was totally useless before anyway, as it was just a dummy response).

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Logging.rst [hunk 6]
```diff
@@ -107,16 +107,18 @@ You might already have stumbled across such an exception (though hopefully not!)
...
   PHP PROCESS:
   Inode: 
   PID: 2296
   UID: 1
   GID: 1
   User: 
 
+.. note::
+
+  As of Flow 7.1 you can configure if the HTTP Request should be logged in the default exception handler via the setting ``Neos.Flow.log.throwables.renderRequestInformation``.
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Logging.rst:119]:
```suggestion
  As of Flow 7.1 you can configure if the HTTP Request should be logged in the default ThrowableStorage via the boolean setting ``Neos.Flow.log.throwables.renderRequestInformation``.
```

**Pull request merged**